src/interpreter/executors/main.rs

```diff
@@ -135,23 +140,20 @@ where
             return Err(InterpreterError::Panic(PanicReason::ContractNotInInputs));
         }
 
-        let storage = &mut self.storage;
         storage
             .deploy_contract_with_id(salt, storage_slots, &contract, &root, &id)
             .map_err(InterpreterError::from_io)?;
 
-        // TODO: Maybe the code below is useless because we can't spend any gas and can't execute
-        //  any code, so the output also should be the same.
         let remaining_gas = create.limit();
         Self::finalize_outputs(
-            &mut create,
+            create,
             false,
             remaining_gas,
             &initial_balances,
             &RuntimeBalances::from(initial_balances.clone()),
-            &self.params,
+            params,
         )?;
-        Ok(create)
+        Ok(())
```

Comment:
[nit] consider consolidating in-memory updating of outputs here

```suggestion
        self.update_transaction_outputs()?;
        Ok(())
```